### ISSUE TITLE
Pulumi 3.78.1 error: confirmation cancelled, not proceeding with the update: Unexpected Escape Sequence: ['\x1b' 'O']

### ISSUE THREAD

**Issue by mevatron** [pulumi/pulumi#13733]
### What happened?

When trying to run `pulumi up -s test` on macOS Monterey, I get the following error when pressing the up arrow from `no` to the `yes` option:

```
error: confirmation cancelled, not proceeding with the update: Unexpected Escape Sequence: ['\x1b' 'O']
```

This does not happen in the default terminal only in the JetBrains Rider terminal emulator with the following shell variables:

```
COMMAND_MODE=unix2003
TERMINAL_EMULATOR=JetBrains-JediTerm
TERM=xterm-256color
FIG_JETBRAINS_SHELL_INTEGRATION=1
SHELL=/bin/bash
LC_CTYPE=UTF-8
```

### Expected Behavior

Should switch from default `no` option to `yes` upon pressing the up arrow.

### Steps to reproduce

Run `pulumi up` without `-y`.

### Output of `pulumi about`

```
Backend        
Name           pulumi.com
URL            https://app.pulumi.com/myusername
User           myusername
Organizations  myusername

Dependencies:
NAME             VERSION
ediri.Vultr      2.15.1
Mono.Unix        7.1.0-final.1.21458.1
Newtonsoft.Json  13.0.3
Pulumi           3.56.0
Pulumi.Command   0.8.1
Pulumi.Random    4.13.2
Pulumi.Tls       4.10.0

Pulumi locates its logs in /var/folders/f7/kc7b85595t53hhlv7szzzgfnry9hxd/T/ by default
```

### Additional context

Working around the issue with `pulumi up -s test -y`.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Zaid-Ajaj** [comment]
Hi @mevatron thank you for filing the issue 🙏 this issue seems to come from the go-survey library we use in the CLI, however, this seems to have been [fixed](https://github.com/go-survey/survey/issues/341) in that library a while ago. We simply might need to update go-survey to a version that has this issue fixed. I will discuss this with the team 😄 

It is worth mentioning that I could not reproduce this from the JetBrains Rider terminal on macOS Ventura v13.2.1 so either latest Rider has fixed it or the OS itself. Any chance you can update JetBrains in the meantime if it isn't latest?

**Comment by mevatron** [comment]
Hi @Zaid-Ajaj! It seems to be intermittent for me as well. I still haven't figured out the exact series of steps that cause the CLI to throw this error, but I can say it works sometimes for me with no issue, and others it immediately fails. I'll keep you posted if I learn more! 

This is also happening on the Latest Rider release `Build #RD-232.9559.61, built on August 22, 2023`.

**Comment by justinvp** [comment]
This can happen when a `dotnet` command like `dotnet build` is run. See https://github.com/dotnet/sdk/issues/15243.

I believe this has now been fixed with https://github.com/pulumi/pulumi/pull/13979, which upgraded the library we use to a more recent version which includes [this fix](https://github.com/go-survey/survey/issues/341). This was included in [v3.85.0](https://github.com/pulumi/pulumi/releases/tag/v3.85.0). Therefore, I'm going to close this as fixed.

Please open a new issue if you're still seeing the problem with Pulumi v3.85.0 or higher.
